daemon/main.go

```diff
@@ -127,6 +132,27 @@ func main() {
 	}
 }
 
+func setupPidFile() {
+	pidfile.SetPidfilePath(defaults.PidFilePath)
+	if err := pidfile.Write(); err != nil {
+		log.WithError(err).Fatal("could not create pidfile")
+	}
+	cleanUp()
+}
+
+func cleanUp() {
```

Comment:
I would rename this to something like initCleanup and also call it in the main function, right after `setupPidFile`. This is mostly because it has longer lived side-effects (the signal handling goroutine) that are a bit more global.

Reply:
Looks better this way.